tests/moduletool/test_release.py

```diff
@@ -140,7 +144,7 @@ def get_changelog_content(after_release: bool) -> str:
     assert gitprovider.get_version_tags(repo=path_module) == expected_tags
     # Verify version
     mod = Module.from_path(path_module)
-    assert mod.version == Version("1.2.4.dev0")
+    assert mod.version == Version("1.2.3.1dev0") if use_four_digits and not v1_module else Version("1.2.4.dev0")
```

Comment:
I would expect `1.2.3.1.dev0` here

Reply:
indeed, I didn't pay attention to this. I think this means there is a bug in existing code. @sanderr could you confirm?

Reply:
there is no bug in the existing code. I find it weird but `Version("1.2.3.1dev0") == Version("1.2.3.1.dev0")`

Reply:
It might be, but I'm not entirely sure we want to stick with it. Please leave this thread unresolved for now as a reminder that I look into it.

Reply:
This looks good to me as it is now, but considering this new information I do think we should have at least one test that asserts the string as it is present in the metadata. Do you know if we already have such a test?